src/auth/src/errors.rs

```diff
@@ -88,3 +89,34 @@ impl Display for CredentialError {
 pub enum InnerAuthError {
     // TODO(#389) - define error types here
 }
+
+#[allow(dead_code)] // TODO(#442) - implementation in progress
+pub(crate) fn is_retryable(c: StatusCode) -> bool {
+    c == StatusCode::INTERNAL_SERVER_ERROR
```

Comment:
`INTERNAL_SERVER_ERROR` gave me pause... why is Okay to retry those?  If you have a good reason, just write a comment so future readers are not confused.

Reply:
Server could have crashed or thrown exception for something not related to the specific request itself right?

If server thinks our request is bad and we should not retry, it should send us 4xx.